design/deployment_design.md

```diff
@@ -0,0 +1,70 @@
+
+Currently, our Python CLI can be deployed in the following ways:
+
+1. Install the CLI via the pip Python package manager like so: `pip3 install conjur-client`. If installed in this way, Python **is** required on machine. If installed via pip, the user can then run the CLI like so `conjur-cli --help`
+
+2. Install and run the executable (Python **is** **not** required on the machine)
+
+   The CLI is delivered as a standalone executable, packed with the required runtime and dependencies. Therefore, there is no need for Python or other prerequisites to be installed on machine. The executable can be fetched by navigating to our [Python CLI release page](https://github.com/cyberark/conjur-api-python3/releases/tag/v0.0.5). If installed in this manner, `./python-cli-executable-location variable get ...`
+
+### Design
+
+#### Pip3 package
+
+We make CLI package available to the public by pushing to a public repository called PyPi. I propose that we maintain pushing our the Python CLI to pip3 for both customer's machines have have internet access and for our development work. The infrastructure is already in-place so this would be at no additional effort.
+
+#### Standalone executable
+
+To install the executable without an internet connection, there are a few options that we can pursue. We can give each executable a different name depending on the intended platform. 
+
+```
+conjur.exe (windows)
+conjur (linux)
+conjur-darwin (macOS)
+```
+
+For Windows, executables have the `.exe` in their name so we will follow suit for the CLI Windows executable. 
+
+For Linux and macOS, we will need to differentiate them. For Linux, we can call the executable `conjur` and for macOS, `conjur-darwin`. That way, the common user-case (Linux) will have a cleaner experience. 
+
+A con of this approach would be that on the [release page](https://github.com/cyberark/conjur-api-python3/releases), at face value, the user will not be able to tell which executable is for linux machines. They will need to make an assumption because "linux" is not in the name as with macOS (-darwin) and Window (.exe) executables.
+
+Another option would be pushing zipped folders for each platform to the release page for our Github repository. Each zipped folder will have its respective executable as detailed below.
+
+```
+linux/conjur-cli.zip
+	conjur
+darwin/conjur-cli.zip
+  conjur
+windows/conjur-cli.zip
+  conjur.exe
+```
+
+Once the user has installed the execuable, they can run the executable in one of the following ways:
+
+1. Run the executable like so: `./linux/conjur-cli/conjur --help` 
+2. Move the ***only*** executable (`conjur`) to their `/usr/local/bin` path so that it would look something like `/usr/local/bin/conjur`. That way, the CLI executable will be available from anywhere on their machine without having to detail the full path
+
+TODO: See if part of the installation process we can automatically save executable to the `/usr/local/bin` directory.
```

Comment:
There is more than one possible location for a binary. It depends on the user and how they work. I think the simplest would be letting the user decide where to unzip it and how to use it.

Reply:
IMO - We can also direct users to add the CLI path to the `path` env var so they can run it from anywhere without moving it to a location that is already in this env var

Reply:
I am 👌  with that